pkg/cmd/experimental/diagnostics/helpers.go

```diff
@@ -0,0 +1,32 @@
+package diagnostics
+
+import (
+	"fmt"
+
+	"github.com/openshift/origin/pkg/diagnostics/log"
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+// determineRequestedDiagnostics determines which diagnostic the user wants to run
+// based on the -d list and the available diagnostics.
+// returns ok (can proceed), warnings, errors, diagnostics
+func determineRequestedDiagnostics(available []string, requested []string, logger *log.Logger) (bool, []error, []error, []string) {
```

Comment:
I don't see this function being re-used anywhere else, plus it is unexported. Does this warrant a separate file. I would move it after its caller (`buildPodDiagnostics`).